### PR TITLE
pfCharts: Component Conversion

### PR DESCRIPTION
**Pull request by dtaylor113**
Updated the following charts:

- pfC3Chart
- pfDonutPctChart
- pfLineChart
- pfSparklineChart
- pfHeatMap
- pfHeatmapLegend
- pfTrendsChart
- pfUtilizationBarChart
- pfUtilizationTrendChart
- pfEmptyChart

Only issue was that pfHeatMap still uses the following to compile it's tooltips:
```
      //Compiles the tooltips
      angular.forEach(angular.element(blocks), function (block) {
        var el = angular.element(block);
        // TODO: get heatmap tooltips to work without using $compile or $scope
        $compile(el)($scope);
      });
```
Will address in a follow up story: [PTNFLY-2068](https://patternfly.atlassian.net/browse/PTNFLY-2068)

### CODE REVIEW

File: src/charts/c3/c3-chart.component.js [hunk 1]
```diff
@@ -64,6 +64,7 @@
        }
 
        $scope.submitform = function (val) {
+         console.log("submitform");
```
Comment (jeff-phillips-18) [inline src/charts/c3/c3-chart.component.js]: Please remove console.log

File: src/charts/c3/c3-chart.component.js [hunk 2]
```diff
@@ -75,32 +76,45 @@
...
+        $log.info("  C3 generateChart");
+
+        // Need to deep watch changes in chart config
+        prevConfig = angular.copy(ctrl.config);
+
+        $timeout(function () {
+          chartData = ctrl.config;
+          if (chartData) {
+            chartData.bindto = '#' + ctrl.id;
+            $log.info("    -> C3 chart generatd: " + chartData.bindto);
```
Comment (jeff-phillips-18) [inline src/charts/c3/c3-chart.component.js]: is it necessary to $log this?

File: src/charts/c3/c3-chart.component.js [hunk 3]
```diff
@@ -75,32 +76,45 @@
...
-                scope.getChartCallback(chart);
-              }
-            }
-          });
-        }, true);
-      }
-    };
+  angular.module('patternfly.charts').component('pfC3Chart', {
+    bindings: {
+      id: '@?',
```
Comment (jeff-phillips-18) [inline src/charts/c3/c3-chart.component.js]: Do we need to bind to id or can we just inject $attrs and get it from there?

File: src/charts/heatmap/heatmap-legend.component.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        ctrl.legendColors = heatmapColorPatternDefaults;
+      }
+      if (!ctrl.legend) {
+        ctrl.legend = legendLabelDefaults;
+      }
+      for (index = ctrl.legend.length - 1; index >= 0; index--) {
+        items.push({
+          text: ctrl.legend[index],
+          color: ctrl.legendColors[index]
+        });
```
Comment (jeff-phillips-18) [inline src/charts/heatmap/heatmap-legend.component.js]: I believe legendItems would need to be updated if ctrl.legend changes. Looks like an existing issue.

File: src/charts/heatmap/heatmap.html [hunk 5]
```diff
@@ -1,9 +1,9 @@
...
-  <div class="heatmap-container" ng-style="containerStyles">
+  <h3>{{$ctrl.chartTitle}}</h3>
+  <div class="heatmap-container" ng-style="$ctrl.containerStyles">
     <svg class="heatmap-pf-svg"></svg>
   </div>
-  <div pf-empty-chart ng-if="chartDataAvailable === false" chart-height="height"></div>
-  <div ng-if="!loadingDone" class="spinner spinner-lg loading"></div>
-  <div ng-if="showLegend" pf-heatmap-legend legend="legendLabels" legend-colors="heatmapColorPattern"></div>
+  <pf-empty-chart ng-if="$ctrl.chartDataAvailable === false" chart-height="$ctrl.height"></pf-empty-chart>
+  <!--div ng-if="!$ctrl.loadingDone" class="spinner spinner-lg loading"></div-->
```
Comment (jeff-phillips-18) [inline src/charts/heatmap/heatmap.html]: Leftover comment?

File: src/charts/trends/trends-chart.html [hunk 6]
```diff
@@ -1,59 +1,59 @@
...
       <div class="col-sm-4 col-md-4 trend-flat-col">
         <div class="trend-flat-details">
           <div class="trend-flat-details-cell">
-            <span class="trend-title-flat-big-pf">{{getPercentageValue() + '%'}}</span>
+            <span class="trend-title-flat-big-pf">{{$ctrl.getPercentageValue() + '%'}}</span>
           </div>
           <div class="trend-flat-details-cell">
-            <span class="trend-label-flat-strong-pf">{{config.trendLabel}}</span>
-            <span class="trend-label-flat-pf">{{getLatestValue()}} of {{chartData.total + ' ' + config.units}}</span>
+            <span class="trend-label-flat-strong-pf">{{$ctrl.nfig.trendLabel}}</span>
```
Comment (jeff-phillips-18) [inline src/charts/trends/trends-chart.html]: should this be $ctrl.config?

File: src/charts/utilization-trend/utilization-trend-chart.html [hunk 7]
```diff
@@ -1,23 +1,23 @@
...
-    <h1 class="available-count pull-left">{{currentValue}}</h1>
+    <h1 class="available-count pull-left">{{$ctrl.currentValue}}</h1>
     <div class="available-text pull-left">
       <div>
-        <span>{{currentText}}</span>
+        <span>{{$ctrl.currentText}}</span>
       </div>
       <div>
-        <span> of {{chartData.total}} {{config.units}}</span>
+        <span> of {{$ctrl.hartData.total}} {{$ctrl.config.units}}</span>
```
Comment (jeff-phillips-18) [inline src/charts/utilization-trend/utilization-trend-chart.html]: This should be $ctrl.chartData

File: src/charts/trends/trends-chart.html (hunk 6 above)
Comment (dtaylor113) [inline src/charts/trends/trends-chart.html]: Wow, great catch!

File: src/charts/utilization-trend/utilization-trend-chart.html (hunk 7 above)
Comment (dtaylor113) [inline src/charts/utilization-trend/utilization-trend-chart.html]: Wow, another great catch, thanks!

**Comment by dtaylor113** [comment]
Hi @jeff-phillips-18 ,
Thanks for the review.  Last commit addresses all review issues.
- Dave

**Pull request merged**